### PR TITLE
distutils-r1.eclass: support >=dev-util/maturin-0.13.0

### PR DESCRIPTION
**Pull request by ionenwks**
@gentoo/python please ack for distutils-r1.eclass changes (see commit message for details), feels rather local/simple (2 revdeps) so won't bother sending to ML unless want that anyway.

If want to confirm passed args, there is a `Running 'maturin ... --compatibility off <eclass-args>'` line in adblock/watchfiles build.log.

On a side-note, most maturin pyproject.toml have a `<0.13` but it doesn't stop use of 0.13 and I know of no issues at the moment.

### CODE REVIEW

File: dev-util/maturin/maturin-0.13.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,419 @@
...
+	local skip=(
+		--skip locked_doesnt_build_without_cargo_lock
+		# move below when >=rust-1.62 is stable, xfail "pass" with 1.60-1.61
+		--skip pyo3_no_extension_module
+	)
+	[[ ${EPYTHON} == pypy3 ]] && skip+=(
+		# test enables pyo3's auto-initialize that is incompatible with pypy
+		--skip integration_pyo3_bin
+		# wants the missing libpypy*-c.so
+#		--skip pyo3_no_extension_module
```
Comment (mgorny) [inline dev-util/maturin/maturin-0.13.0.ebuild:408]: Is there a reason to keep it if it's commented out?

File: eclass/distutils-r1.eclass [hunk 2]
```diff
@@ -1442,12 +1442,20 @@ distutils-r1_python_compile() {
...
+				local -x MATURIN_PEP517_ARGS="
+					--jobs=$(makeopts_jobs)
+					--skip-auditwheel
+					$(in_iuse debug && usex debug --profile=dev '')
+				"
+			else
+				# legacy support, can cleanup when depend on >=0.13
+				local -x MATURIN_PEP517_ARGS=--skip-auditwheel
+				in_iuse debug && use debug &&
+					MATURIN_PEP517_ARGS+=" --cargo-extra-args=--profile=dev"
```
Comment (mgorny) [inline eclass/distutils-r1.eclass:1457]: I suppose you could use the `$()` style here too for consistency/reaadability.

File: dev-util/maturin/maturin-0.13.0.ebuild (hunk 1 above)
Comment (ionenwks) [reply dev-util/maturin/maturin-0.13.0.ebuild:408]: It's just a self-note to uncomment it when drop it above.

File: eclass/distutils-r1.eclass (hunk 2 above)
Comment (ionenwks) [reply eclass/distutils-r1.eclass]: Sure I guess, should be short lived but made it the same as above.

**Comment by mgorny** [comment]
@ionenwks, could you add the ebegin patch (v2) from gentoo-dev on top of it? Feel free to merge afterwards.

**Review by thesamesam — APPROVED** [review]

**Comment by mgorny** [comment]
Could you also bump Python 3.11 dep to beta4 in python-utils-r1?

**Comment by ionenwks** [comment]
> Could you also bump Python 3.11 dep to beta4 in python-utils-r1?

done, let me know if still want ML's check_deps using v3 or just leave it as ebegin/end

**Comment by mgorny** [comment]
Let's leave it as-is for now. We can revisit when we're removing old maturin support.

**Pull request closed without merging**